### ISSUE TITLE
Watchdog timeout not honored

### ISSUE THREAD

**Issue by CapOM** [systemd/systemd#9467]
**systemd version the issue has been seen with**
237

**Used distribution**
 > Ubuntu 18.04

**Expected behaviour you didn't see**
 > WatchdogSec not honored

**Unexpected behaviour you saw**
 > Watchdog timeout triggered right upon running systemctl daemon-reload

**Steps to reproduce the problem**

1) Create a service with:
```
[Service]
NotifyAccess=all
ExecStart=/bin/sleep 120
```
2) Then run it with systemctl start test
3) Edit test.service and add: (in a real case it is done with a drop-in)
```
Type=notify
WatchdogSec=30s
```
4) Load the new config with systemctl daemon-reload

Actual result: watchdog timeout is triggered less than 1 second after running systemctl daemon-reload

Expected resukt: watchdog timeout should happen after 30s, i.e. the configured timeout.

**Comment by boucman** [comment]
just to make sure... it is the watchdog timout that is triggered, not another watchdog ?

I am asking because the Type=notify means that your service is staying indefinitely in the "activating" phase, and that phase has its own timeouts...

**Comment by CapOM** [comment]
Yes:

Jun 28 14:12:15 julien-virtual-machine systemd[1]: Reloading.
Jun 28 14:12:15 julien-virtual-machine systemd[1]: test.service: Watchdog timeout (limit 30s)!

**Comment by michaelolbrich** [comment]
I don't think it makes sense to enable the timeout after a service has been started. How should the application know, that the timeout changed?
That would break any update that introduces or changes the timeout for any service that is not immediately restarted.

**Comment by poettering** [comment]
hmm, we should probably not honour watchdog time changes until the next reload, as otherwise $WATCHDOG_USEC will be out of sync with what we actually expect.

**Comment by poettering** [comment]
so yeah, i figure we should copy the watchdog timeout configured at the time we start the service and stick to that exclusively.

**Comment by poettering** [comment]
Fix waiting in #10508.

**Comment by poettering** [comment]
(this fix will rearrange things so that the watchdog setting in effect at the moment of fork() remains in effect during the whole cycle. Only this way we can make sure $WATCHDOG_USEC remains in sync with what we expect ourselves.
